wandb/cli/cli.py

```diff
@@ -1519,6 +1515,183 @@ def scheduler(
         raise e
 
 
+@cli.group("job")
+def job() -> None:
+    pass
+
+
+@job.command("list")
+@click.option(
+    "--project",
+    "-p",
+    envvar=env.PROJECT,
+    help="The project you want to list jobs from.",
+)
+@click.option(
+    "--entity",
+    "-e",
+    default="models",
+    envvar=env.ENTITY,
+    help="The entity the jobs belong to",
+)
+def _list(project, entity):
+    wandb.termlog(f"Listing jobs in {entity}/{project}")
+
+    public_api = PublicApi()
+    try:
+        jobs = public_api.list_jobs(entity=entity, project=project)
+    except wandb.errors.CommError:
+        wandb.termerror(f"Error listing jobs for entity/project: {entity}/{project}")
+        return
+
+    for job in jobs:
+        for version in job["edges"]:
+            name = version["node"]["artifactSequence"]["name"]
+            full_name = f"{entity}/{project}/{name}"
+            aliases = [x["alias"] for x in version["node"]["aliases"]]
+            if len(aliases) == 1:
+                # no custom alias, just verison, skip
+                continue
+            wandb.termlog(f"{full_name} {aliases}")
+
+
+@job.command()
+@click.argument("job")
+def describe(job):
+    public_api = PublicApi()
+    job = public_api.job(name=job)
+
+    for key in job._job_info:
+        if key.startswith("_"):
+            continue
+        wandb.termlog(f"{key}: {job._job_info[key]}")
+
+
+@job.command(
+    no_args_is_help=True,
+)
+@click.option(
+    "--project",
+    "-p",
+    envvar=env.PROJECT,
+    help="The project you want to list jobs from.",
+)
+@click.option(
+    "--entity",
+    "-e",
+    default="models",
+    envvar=env.ENTITY,
+    help="The entity the jobs belong to",
+)
+@click.option(
+    "--name",
+    "-n",
+    help="Name for the job",
+)
+@click.option(
+    "--description",
+    "-d",
+    help="Description for the job",
+)
+@click.option(
+    "--alias",
+    "-a",
+    "aliases",
+    help="Alias for the job",
+    multiple=True,
+    default=tuple(),
+)
+@click.option(
+    "--entry-point",
+    "-E",
+    "entrypoint",
+    help="Codepath to the main script, required for repo jobs",
+)
+@click.option(
+    "--git-hash",
+    "-g",
+    "git_hash",
+    type=str,
+    help="Hash to a specific git commit.",
+)
+@click.option(
+    "--runtime",
+    "-r",
+    type=str,
+    help="Python runtime to execute the job",
+)
+@click.argument(
+    "job_type",
+    type=click.Choice(("repo", "artifact", "image")),
+)
+@click.argument("path")
+def create(
+    path,
+    project,
+    entity,
+    name,
+    job_type,
+    description,
+    aliases,
+    entrypoint,
+    git_hash,
+    runtime,
+):
+    """Create a job from a source, without a wandb run.
+
+    Jobs can be of three types, repo, artifact, or image.
```

Comment:
The names of the three job types here–"repo, artifact, image"–were confusing to me, for a couple of reasons:

- Image jobs can be specified by a path to an image repository like ECR, so at first glance I thought "repo" was what I should use for Docker jobs (thinking of image repositories rather than git repositories)
- The types of jobs here don't align with the [three types of jobs](https://docs.wandb.ai/guides/launch/create-job#how-do-i-create-a-job) we list in our docs

Would it seem worth it to align these types with the Git|Code|Docker job names from the docs?

Reply:
@KyleGoyette Hoff has a point, I have aligned the names based on our SDK code, but these don't match up with the docs...

Reply:
Putting my customer hat on, "Artifact" feels confusing to me since we teach customers that Artifacts are typically Datasets and Models, rather than code they'd log to us (the fact that the code is stored as an artifact is more of an implementation detail that users shouldn't need to worry about).

Could we do "Code" or "Code_Artifact" instead?

Git instead of Repo is 💯, though!

Reply:
@HipHoff  Done